src/FrequenciesTrait.php

```diff
@@ -7,358 +7,388 @@
+	 */
+	public function everyThirtyMinutes()
+	{
+		$this->expression['min']  = '/30';
+		$this->expression['hour'] = '*';
+
+		return $this;
+	}
+
+	/**
+	 * Runs every Sunday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everySunday(string $time = null)
+	{
+		return $this->setDayOfWeek(0, $time);
+	}
+
+	/**
+	 * Runs every monday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everyMonday(string $time = null)
+	{
+		return $this->setDayOfWeek(1, $time);
+	}
+
+	/**
+	 * Runs every Tuesday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everyTuesday(string $time = null)
+	{
+		return $this->setDayOfWeek(2, $time);
+	}
+
+	/**
+	 * Runs every Wednesday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everyWednesday(string $time = null)
+	{
+		return $this->setDayOfWeek(3, $time);
+	}
+
+	/**
+	 * Runs every Thursday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everyThursday(string $time = null)
+	{
+		return $this->setDayOfWeek(4, $time);
+	}
+
+	/**
+	 * Runs every Friday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everyFriday(string $time = null)
+	{
+		return $this->setDayOfWeek(5, $time);
+	}
+
+	/**
+	 * Runs every Saturday at midnight, unless time passed in.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function everySaturday(string $time = null)
+	{
+		return $this->setDayOfWeek(6, $time);
+	}
+
+	/**
+	 * Should run the first day of every month.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function monthly(string $time = null)
+	{
+		$min = $hour = 0;
+
+		if (! empty($time))
+		{
+			[
+				$min,
+				$hour,
+			] = $this->parseTime($time);
+		}
+
+		$this->expression['min']        = $min;
+		$this->expression['hour']       = $hour;
+		$this->expression['dayOfMonth'] = 1;
+
+		return $this;
+	}
+
+	/**
+	 * Should run the first day of each quarter,
+	 * i.e. Jan 1, Apr 1, July 1, Oct 1
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function quarterly(string $time = null)
+	{
+		$min = $hour = 0;
+
+		if (! empty($time))
+		{
+			[
+				$min,
+				$hour,
+			] = $this->parseTime($time);
+		}
+
+		$this->expression['min']        = $min;
+		$this->expression['hour']       = $hour;
+		$this->expression['dayOfMonth'] = 1;
+		$this->expression['month']      = '/3';
+
+		return $this;
+	}
+
+	/**
+	 * Should run the first day of the year.
+	 *
+	 * @param string|null $time
+	 *
+	 * @return $this
+	 */
+	public function yearly(string $time = null)
+	{
+		$min = $hour = 0;
+
+		if (! empty($time))
+		{
+			[
+				$min,
```

Comment:
Actually - this is really annoying and hard to read. Is there a way to turn this rule off?

Reply:
I don't let it run automatically bc I think it is too pushy. I will revert these ones.